presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -157,11 +157,26 @@ public SemiTransactionalHiveMetastore(
     }
 
     public synchronized Optional<Table> getTable(String databaseName, String tableName)
+    {
+        return getTable(databaseName, tableName, Optional.empty());
+    }
+
+    public synchronized Optional<Table> getTable(String databaseName, String tableName, Optional<ConnectorSession> session)
```

Comment:
@arhimondr is this the right place to do this check to discard/enable semantic types?

This seemed like a good compromise because 
* it's above `CachingHiveMetastore` (I imagine that session param checks below the caching layer may have unpredictable results)
* but below classes that issue many calls to this method, such as `HiveMetadata`, `HiveSplitManager`, `HivePartitionManager`...

Reply:
From what I remember Table is not the only object containing the column type informations. Partition object also includes column type information.

Reply:
Alternatively , can we just make the third parameter a boolean and states `discardSemanticType` instead of passing the whole `session` and only use one boolean there?